code/game/objects/structures/mannequin.dm

```diff
@@ -0,0 +1,1187 @@
+		if(SLOT_MANNEQUIN_GLOVES)
+			slotIcon[MANNEQUIN_ICONS_SLOT] = 'icons/mob/hands.dmi'
+			slotIcon[MANNEQUIN_ICONS_PRIMITIVE] = 'icons/mob/hands.dmi'
+			slotIcon[MANNEQUIN_ICONS_SPECIES] = species.gloves_icons
+			slotIcon[MANNEQUIN_DYNAMIC_LAYER] = GLOVES_LAYER
+		if(SLOT_MANNEQUIN_EARS)
+			slotIcon[MANNEQUIN_ICONS_SLOT] = 'icons/mob/ears.dmi'
+			slotIcon[MANNEQUIN_ICONS_PRIMITIVE] = 'icons/mob/ears.dmi'
+			slotIcon[MANNEQUIN_ICONS_SPECIES] = species.ears_icons
+			slotIcon[MANNEQUIN_DYNAMIC_LAYER] = EARS_LAYER
+		if(SLOT_MANNEQUIN_OCLOTHING)
+			slotIcon[MANNEQUIN_ICONS_SLOT] = 'icons/mob/suit.dmi'
+			slotIcon[MANNEQUIN_ICONS_PRIMITIVE] = 'icons/mob/suit.dmi'
+			slotIcon[MANNEQUIN_ICONS_SPECIES] = species.wear_suit_icons
+			slotIcon[MANNEQUIN_ICONS_FAT] = species.fat_wear_suit_icons
+			slotIcon[MANNEQUIN_DYNAMIC_LAYER] = SUIT_LAYER
+		if(SLOT_MANNEQUIN_EYES)
+			slotIcon[MANNEQUIN_ICONS_SLOT] = 'icons/mob/eyes.dmi'
+			slotIcon[MANNEQUIN_ICONS_PRIMITIVE] = 'icons/mob/monkey_eyes.dmi'
+			slotIcon[MANNEQUIN_ICONS_SPECIES] = species.glasses_icons
+			slotIcon[MANNEQUIN_DYNAMIC_LAYER] = GLASSES_LAYER
+		if(SLOT_MANNEQUIN_BELT)
+			slotIcon[MANNEQUIN_ICONS_SLOT] = 'icons/mob/belt.dmi'
+			slotIcon[MANNEQUIN_ICONS_PRIMITIVE] = 'icons/mob/belt.dmi'
+			slotIcon[MANNEQUIN_ICONS_SPECIES] = species.belt_icons
+			slotIcon[MANNEQUIN_DYNAMIC_LAYER] = BELT_LAYER
+		if(SLOT_MANNEQUIN_MASK)
+			slotIcon[MANNEQUIN_ICONS_SLOT] = 'icons/mob/mask.dmi'
+			slotIcon[MANNEQUIN_ICONS_PRIMITIVE] = 'icons/mob/monkey.dmi'
+			slotIcon[MANNEQUIN_ICONS_SPECIES] = species.wear_mask_icons
+			slotIcon[MANNEQUIN_DYNAMIC_LAYER] = FACEMASK_LAYER
+		if(SLOT_MANNEQUIN_HEAD)
+			slotIcon[MANNEQUIN_ICONS_SLOT] = 'icons/mob/head.dmi'
+			slotIcon[MANNEQUIN_ICONS_PRIMITIVE] = 'icons/mob/monkey_head.dmi'
+			slotIcon[MANNEQUIN_ICONS_SPECIES] = species.head_icons
+			slotIcon[MANNEQUIN_DYNAMIC_LAYER] = HEAD_LAYER
+		if(SLOT_MANNEQUIN_BACK)
+			slotIcon[MANNEQUIN_ICONS_SLOT] = 'icons/mob/back.dmi'
+			slotIcon[MANNEQUIN_ICONS_PRIMITIVE] = 'icons/mob/back.dmi'
+			slotIcon[MANNEQUIN_ICONS_SPECIES] = species.back_icons
+			slotIcon[MANNEQUIN_DYNAMIC_LAYER] = BACK_LAYER
+		if(SLOT_MANNEQUIN_ID)
+			slotIcon[MANNEQUIN_ICONS_SLOT] = 'icons/mob/ids.dmi'
+			slotIcon[MANNEQUIN_ICONS_PRIMITIVE] = 'icons/mob/ids.dmi'
+			slotIcon[MANNEQUIN_DYNAMIC_LAYER] = ID_LAYER
+	return slotIcon
+
+/obj/structure/mannequin/proc/get_bloodsies_state(var/slot, var/obj/item/clothToWear)
+	switch(slot)
+		if(SLOT_MANNEQUIN_ICLOTHING)
+			return "uniformblood"
+		if(SLOT_MANNEQUIN_FEET)
+			return "shoeblood"
+		if(SLOT_MANNEQUIN_GLOVES)
+			return "bloodyhands"
+		if(SLOT_MANNEQUIN_OCLOTHING)
+			var/obj/item/clothing/suit/C = clothToWear
+			return "[C.blood_overlay_type]blood"
+		if(SLOT_MANNEQUIN_MASK)
+			return "maskblood"
+		if(SLOT_MANNEQUIN_HEAD)
+			return "helmetblood"
+		else
+			return null
+
+/obj/structure/mannequin/proc/checkMappingWear()
+	if(mapping_uniform)
```

Comment:
I feel like there's a way this COULD be fully softcoded. But at the least you could simply do...
var/list/stuff = list(slot_mannequin_iclothing = mapping_uniform, slot_mannequin_feet = mapping_shoes,...)

and then simply loop through that with an EXTREMELY short loop with almost zero copy pasting.